app/uk/gov/hmrc/personaldetailsvalidation/views/pages/PersonalDetailsPage.scala

```diff
@@ -40,11 +41,44 @@ private[personaldetailsvalidation] class PersonalDetailsPage @Inject()(implicit
   private val form: Form[PersonalDetails] = Form(mapping(
     "firstName" -> mandatoryText("personal-details.firstname.required"),
     "lastName" -> mandatoryText("personal-details.lastname.required"),
-    "nino" -> mandatoryText("personal-details.nino.required")
-      .verifying("personal-details.nino.invalid", nonEmptyString => Try(Nino(nonEmptyString.value)).isSuccess)
-      .transform[Nino](validatedNonEmptyNino => Nino(validatedNonEmptyNino.value), nino => NonEmptyString(nino.toString)),
-    "dateOfBirth" -> mandatoryLocalDate("personal-details")
-  )(PersonalDetails.apply)(PersonalDetails.unapply))
+    "nino" -> optionalText
+      .verifying("personal-details.nino.invalid", nonEmptyString => nonEmptyString.forall(nonEmptyString => Try(Nino(nonEmptyString.value)).isSuccess))
+      .transform[Option[Nino]](_.map(nonEmptyString => Nino(nonEmptyString.value)), _.map(nino => NonEmptyString(nino.value))),
+    "dateOfBirth" -> mandatoryLocalDate("personal-details"),
+    "postcode" -> optionalText
+      .verifying("personal-details.postcode.invalid", nonEmptyString => nonEmptyString.forall(nonEmptyString => Try(nonEmptyString.value).isSuccess))
```

Comment:
nonEmptyString => nonEmptyString.forall(_ => true) Can be used instead of this long check
But in this case leave it for the next story to validate the postcode.